### ISSUE TITLE
model.evaluate() gives less accuracy as 0.1 for fashion mnist dataset in keras

### ISSUE THREAD

**Issue by sksatgithub** [keras-team/keras#13655]
I am using google colab. Here is the full code which is exactly same as on the official tensorflow tutorial site. But the problem with my code is that the model.evaluate() produces very less accuracy value which is 0.1. I am unable to solve this issue. Kindly help.

from __future__ import absolute_import, division, print_function, unicode_literals
#import tenseorflow first
import tensorflow as tf
#from tensorflow import keras for datasets
from tensorflow import keras

#import some helper libraries
import numpy as np
import matplotlib.pyplot as plt

#first load dataset of fashion mnist from keras
my_fashion_mnist = keras.datasets.fashion_mnist
(train_images,train_labels),(test_images,test_labels) = my_fashion_mnist.load_data()

#define name of all categories
image_category = ['tshirt/top','trouser','pullover','dress','coat','sandal','shirt','sneaker','bag','ankel boot']

#now change all the training images into grayscale
trian_images = train_images/255.0

#also change all the test images into grayscale
test_images = test_images/255.0

#display first 25 images befor training the model
plt.figure(figsize=(10,10))
for i in range(25):
  plt.subplot(5,5,i+1)
  plt.xticks([])
  plt.yticks([])
  plt.grid(False)
  plt.imshow(train_images[i],cmap=plt.cm.binary)
  plt.xlabel(image_category[train_labels[i]])
plt.show()

#now build the model
model = keras.Sequential([
                          keras.layers.Flatten(input_shape=(28,28)),
                          keras.layers.Dense(128,activation='relu'),
                          keras.layers.Dense(10,activation='softmax')
])

#compile the model
model.compile(optimizer='adam',loss='sparse_categorical_crossentropy',metrics=['accuracy'])

#train the model
model.fit(train_images,train_labels,epochs=10)

#evaluate the model
test_loss,test_accuracy = model.evaluate(test_images,test_labels,verbose=2)
print('test accuracy: ',test_accuracy)

the output is as below:
Train on 60000 samples
Epoch 1/10
60000/60000 [==============================] - 4s 74us/sample - loss: 3.0651 - acc: 0.6951
Epoch 2/10
60000/60000 [==============================] - 4s 72us/sample - loss: 0.6817 - acc: 0.7332
Epoch 3/10
60000/60000 [==============================] - 4s 70us/sample - loss: 0.5925 - acc: 0.7810
Epoch 4/10
60000/60000 [==============================] - 4s 71us/sample - loss: 0.5363 - acc: 0.8143
Epoch 5/10
60000/60000 [==============================] - 4s 71us/sample - loss: 0.5072 - acc: 0.8261
Epoch 6/10
60000/60000 [==============================] - 4s 71us/sample - loss: 0.4979 - acc: 0.8302
Epoch 7/10
60000/60000 [==============================] - 4s 70us/sample - loss: 0.4844 - acc: 0.8349
Epoch 8/10
60000/60000 [==============================] - 4s 73us/sample - loss: 0.4754 - acc: 0.8397
Epoch 9/10
60000/60000 [==============================] - 4s 70us/sample - loss: 0.4696 - acc: 0.8405
Epoch 10/10
60000/60000 [==============================] - 4s 70us/sample - loss: 0.4666 - acc: 0.8429
10000/10000 - 0s - loss: 2.6903 - acc: 0.1029
test accuracy:  0.1029

**Comment by mpaloni** [comment]
There seems to be a typo in the code: 

> #now change all the training images into grayscale
> **trian_images** = train_images/255.0

If that is correct in your source code. The error seems to be somewhere in the test data vs training data - either the data is different somehow, for example the scaling. Another thing that I have faced is that the data is not divided in correct manner, for example testing and training data set do not have same categories..

**Comment by ns3284** [comment]
As mpaloni said, you have a typo, so your train_images are not scaled, and your test_images are. This could be causing your poor performance. Additionally, dividing by 255.0 does not convert it to grayscale, as the comment says, it just beings the 1byte integer values to 0-1 floats.

**Comment by sushreebarsa** [comment]
@sksatgithub  Moving this issue to closed status as there has been no recent activity, in case you still face the error please create a new issue.Thanks!
